app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
      */
-    var wasDefaultBrowserOnLastResume by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = isDefaultBrowser()
-    )
+    fun checkDefaultBrowserAndSet(): Boolean {
+        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
+        val isDefaultBrowserNow = isDefaultBrowser()
+        val isDefaultBrowserBefore = this.preferences.getBoolean(prefKey, isDefaultBrowserNow)
```

Comment:
**suggestion (non-blocking):** when is "before"? Perhaps we should be more explicit when this is expected to be called (in the function name, e.g. `checkDefaultBrowserAndSetInOnResume`) and represent that in the this variable name (e.g. `wasDefaultBrowserOnLastResume`). I know we named these for the `!=` logic below, but given the logic is different than I thought (see below), it might be better to rename.